lib/src/git.rs

```diff
@@ -2034,24 +2034,24 @@ where
     // detect whether one is present with the current API, and `jj git remote
     // set-url` refuses to work with them anyway.
     new_remote = new_remote.with_fetch_tags(remote.fetch_tags());
-    for direction in [gix::remote::Direction::Fetch, gix::remote::Direction::Push] {
-        new_remote
-            .replace_refspecs(
-                remote
-                    .refspecs(direction)
-                    .iter()
-                    .map(|refspec| refspec.to_ref().to_bstring()),
-                direction,
-            )
-            .expect("existing refspecs to be valid");
-    }
+    new_remote
+        .replace_refspecs(
+            remote
+                .refspecs(gix::remote::Direction::Fetch)
+                .iter()
+                .map(|refspec| refspec.to_ref().to_bstring()),
+            gix::remote::Direction::Fetch,
+        )
+        .expect("existing refspecs to be valid");
     Ok(new_remote)
 }
 
 pub fn set_remote_url(
     store: &Store,
     remote_name: &RemoteName,
     new_remote_url: &str,
+    fetch: bool,
+    push: bool,
```

Comment:
Maybe we can instead add `new_push_url: Option<&str>` or something? It's unlikely we want to set the same URL to both `url` and `pushUrl`.